extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/util/SchedulerUtils.java

```diff
@@ -47,28 +66,64 @@ public static long parseEveryAsMillis(Scheduled scheduled) {
      */
     public static String lookUpPropertyValue(String propertyValue) {
         String value = propertyValue.trim();
-        if (!value.isEmpty() && isConfigValue(value)) {
-            value = ConfigProviderResolver.instance().getConfig().getValue(getConfigProperty(value), String.class);
+        if (!value.isEmpty()) {
+            value = resolvePropertyExpression(adjustExpressionSyntax(value));
         }
-
         return value;
     }
 
     public static boolean isConfigValue(String val) {
+        return isSimpleConfigValue(val) || isConfigExpression(val);
+    }
+
+    private static boolean isSimpleConfigValue(String val) {
         val = val.trim();
         return val.startsWith("{") && val.endsWith("}");
     }
 
-    private static String getConfigProperty(String val) {
-        return val.substring(1, val.length() - 1);
+    /**
+     * Converts "{property}" to "${property}" for backwards compatibility
+     */
+    private static String adjustExpressionSyntax(String val) {
+        if (isSimpleConfigValue(val)) {
+            return '$' + val;
+        }
+        return val;
+    }
+
+    /**
+     * Adapted from {@link io.smallrye.config.ExpressionConfigSourceInterceptor}
+     */
+    private static String resolvePropertyExpression(String expr) {
+        final Config config = ConfigProviderResolver.instance().getConfig();
+        final Expression expression = compileExpression(expr);
+        final String expanded = expression.evaluate((resolveContext, stringBuilder) -> {
```

Comment:
Also we try to avoid using lambdas in runtime modules if possible.

Reply:
Sorry, that was copy/paste from io.smallrye.config.ExpressionConfigSourceInterceptor. 
Done.